google-cloud-datastore/src/test/java/com/google/cloud/datastore/it/ITDatastoreTest.java

```diff
@@ -137,6 +139,46 @@
   private static final Entity ENTITY3 = Entity.newBuilder(ENTITY1).setKey(KEY3).remove("str")
       .set("null", NULL_VALUE).set("partial1", PARTIAL_ENTITY2).set("partial2", ENTITY2).build();
 
+  private <T> Iterator<T> getStronglyConsistentResults (Query scQuery, Query query) {
+    QueryResults<T> scResults = DATASTORE.run(scQuery);
+    List<T> scResultsCopy = makeResultsCopy(scResults);
+    Set<T> scResultsSet = new HashSet<>(scResultsCopy);
```

Comment:
This will work only if the query returns only distinct results. Is it guaranteed to return distinct result items all the time? This is also under assumption, that equals and hascode are properly defined fro the result items.

Please (at least) provide a comment mentioning that the items in the result must be unique.